perceval/simulators/simulator.py

```diff
@@ -397,4 +409,32 @@ def evolve(self, input_state: Union[BasicState, StateVector]) -> StateVector:
             result_sv += evolved_in_s * probampli
 
         result_sv.normalize()
-        return result_sv
+        return self._post_select_on_statevector(result_sv)
+
+    def evolve_svd(self, svd: Union[SVDistribution, StateVector, BasicState], progress_callback: Optional[Callable] = None) -> SVDistribution:
+        """
+        Compute the SVDistribution evolved through a Linear Optical circuit
+
+        :param svd: The input StateVector distribution
+        :param progress_callback: A function with the signature `func(progress: float, message: str)`
+        :return: The output StateVectorDistribution
+
+        """
+        if not isinstance(svd, SVDistribution):
+            return SVDistribution(self.evolve(svd))
+
+        # If it's actually an SVD
+
+        new_svd = SVDistribution()
+        for idx, (sv, p) in enumerate(svd.items()):
+            if min(sv.n) >= self._min_detected_photons:
+                new_sv = self._post_select_on_statevector(self.evolve(sv))
+                new_svd[new_sv] = p
```

Comment:
I think
`new_svd[new_sv] += p` would fix corner cases where `new_sv` could be the result of several post_selected evolutions.